File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int change(string s){
    return ((s[0] - '0') * 10 + (s[1] - '0')) * 60 + (s[3] - '0') * 10 + (s[4] - '0');
}

string re(int num){
    string ss = "     ";
    int m = num / 60;
    int s = num % 60;
    
    ss[0] = m / 10 + '0';
    ss[1] = m % 10 + '0';
    ss[2] = ':';
    ss[3] = s / 10 + '0';
    ss[4] = s % 10 + '0';
    return ss;
}

string solution(string video_len, string pos, string op_start, string op_end, vector<string> commands) {
    
    int cur = change(pos);
    int end = change(video_len);
    int os = change(op_start);
    int oe = change(op_end);
    
    for (string c : commands){
        if (cur >= os && cur <= oe){
            cur = oe;
        }
            
        if (c == "prev"){
            if (cur - 10 <= 0)
                cur = 0;
            else
                cur -= 10;
        }
        else {
            if (cur + 10 >= end)
                cur = end;
            else
                cur += 10;
        }
    }
    
    if (cur >= os && cur <= oe){
        cur = oe;
    }
    
    return re(cur);
}

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def change(time):
    h, m = map(int, time.split(':'))
    return h * 60 + m

def re(time):
    h = time // 60
    m = time % 60
    
    
    return str(h).zfill(2) + ':' + str(m).zfill(2)

def solution(video_len, pos, op_start, op_end, commands):
    video_len = change(video_len)
    pos = change(pos)
    op_start = change(op_start)
    op_end = change(op_end)
    
    for c in commands:
        if op_start <= pos <= op_end:
            pos = op_end
        
        if c == 'next':
            pos += 10
            if pos > video_len:
                pos = video_len
        else:
            pos -= 10
            if pos < 0:
                pos = 0
        
        if op_start <= pos <= op_end:
            pos = op_end
            
    return re(pos)